File: DSA 4th Year/TUF/Patterns/nstar.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{

    int n;
    cin >> n;
    for (int i = 1; i <= n; i++)
    {
        int spaces = i - 1;
        while (spaces != 0)
        {
            cout << " ";
            spaces--;
        }
        for (int j = 1; j <= 2 * n - (2 * i )+ 1; j++)
        {
            cout << "*";
        }
        cout << "\n";
    }

    return 0;
}

File: DSA 4th Year/TUF/Patterns/nstar.py
def nStarTriangle(n: int) -> None:
    for i in range(1,n+1):
        spaces=i-1
        while(spaces!=0):
            print(" ",end="")
            spaces-=1

        for j in range(1,(2*n)-i-1):
            print("*",end="")

        print("\n")



nStarTriangle(10)a

    